apps/server/src/modules/account/domain/services/account.service.ts

```diff
@@ -182,6 +182,23 @@ export class AccountService extends AbstractAccountService {
 		return targetAccount;
 	}
 
+	public async deactivateAccount(userId: EntityId, deactivatedAt: Date): Promise<void> {
+		const account = await this.accountRepo.findByUserIdOrFail(userId);
+		account.deactivatedAt = deactivatedAt;
+
+		await this.save(account);
+	}
+
+	public async reactivateAccount(userId: EntityId): Promise<void> {
+		const account = await this.accountRepo.findByUserIdOrFail(userId);
+		account.deactivatedAt = undefined;
+		await this.save(account);
+	}
+
+	public async deactivateMultipleAccounts(userIds: EntityId[], deactivatedAt: Date): Promise<void> {
+		await this.accountRepo.deactivateMultipleByUserIds(userIds, deactivatedAt);
+	}
```

Comment:
Should it be possible to deactivate an account at a specific time? I would suppose that the operation is just deactivating / reactivating one or more accounts. In that case we wouldn't maybe need the `deactivatedAt` parameter. The timestamp could just be set to the current date internally.
Also would the parameter allow to set a time in tha past. That's maybe not what we want.